Ticket 3089 (Halo node properties in Cylindrical2dMesh)

jmosborne opened the pull request:
Add some member variables to change properties of Halo Nodes in Cylindrical2dMesh. #3089
Also change bounded cylindrical meshes to only have 1 ghost image node for each boundary edge, as more stable #3078

jmosborne committed:
ticket #3089 Added member varibles and get set methods. Also edited bounded cylindrical meshes to only have one ghost node for voronoi construction as more stable

jmosborne committed:
ticket #3089 removing some code commited in error

jmosborne committed:
Ticket #3089 Extend bounded VT to work with voids on Cylindrical2DMesh

jmosborne committed:
Merge branch 'develop' into ticket_3089

jmosborne committed:
Updating archive. Note not with correct Boost.

jmosborne commented:
Need to update archive on a boost 1.55 machine. current archive is generated on 1.70 on docker.

jmosborne committed:
Merge branch 'develop' into ticket_3089

jmosborne committed:
ticket 3089 updating archive

jmosborne commented:
Need to add test of generating a vertex mesh for a bounded VT mesh including where the image nodes are close to existing nodes to complete coverage

jmosborne committed:
Fixing coverage and tidying whitespace.

jmosborne committed:
Tidying whitespace.

fcooper8472 commented on mesh/src/mutable/Cylindrical2dMesh.cpp:
```diff
@@ -160,10 +163,10 @@ void Cylindrical2dMesh::CreateHaloNodes()
     mBottomHaloNodes.clear();
 
-    unsigned num_halo_nodes = (unsigned)(floor(mWidth*2.0));
+    unsigned num_halo_nodes = (unsigned)(floor(mWidth*mHaloScalingFactor));
```
Ideally these old c-style casts should be replaced with `static_cast`, e.g.
```cpp
unsigned num_halo_nodes = static_cast<unsigned>(floor(mWidth*mHaloScalingFactor));
```

jmosborne replied:
Fixed and updated a few more I caught in the classes.

fcooper8472 commented on mesh/src/mutable/Cylindrical2dMesh.cpp:
```diff
@@ -160,10 +163,10 @@ void Cylindrical2dMesh::CreateHaloNodes()
     mBottomHaloNodes.clear();
 
-    unsigned num_halo_nodes = (unsigned)(floor(mWidth*2.0));
+    unsigned num_halo_nodes = (unsigned)(floor(mWidth*mHaloScalingFactor));
```
Needs to have a formatter run over this: e.g. there should be a single space either side of binary operators such as * and /

The style is contained in the .clang-format file in the repo: all main IDEs will auto-format based on that style file.

jmosborne replied:
Fixed this and others I found.

fcooper8472 commented on mesh/src/mutable/Cylindrical2dMesh.cpp:
```diff
@@ -844,6 +847,28 @@ unsigned Cylindrical2dMesh::GetCorrespondingNodeIndex(unsigned nodeIndex)
     return corresponding_node_index;
 }
 
+double Cylindrical2dMesh::GetHaloScalingFactor()
```
Method should be const

jmosborne replied:
Done

fcooper8472 commented on mesh/src/mutable/Cylindrical2dMesh.cpp:
```diff
@@ -844,6 +847,28 @@ unsigned Cylindrical2dMesh::GetCorrespondingNodeIndex(unsigned nodeIndex)
+    mHaloScalingFactor = haloScalingFactor;
+}
+
+double Cylindrical2dMesh::GetHaloOffset()
```
method should be const

jmosborne replied:
Done

fcooper8472 commented on mesh/src/vertex/Cylindrical2dVertexMesh.cpp:
```diff
@@ -150,40 +152,100 @@ Cylindrical2dVertexMesh::Cylindrical2dVertexMesh(Cylindrical2dMesh& rMesh, bool
-                            }
+                    
+                    double ratio = 0.5;
+                    c_vector<double,2> new_node_location = normal_vector + p_node_a->rGetLocation() + ratio*edge;
```
formatting: space either side of *

jmosborne replied:
Done along with rest of the occurrences in the file.

fcooper8472 commented on mesh/src/vertex/Cylindrical2dVertexMesh.cpp:
```diff
@@ -150,40 +152,100 @@ Cylindrical2dVertexMesh::Cylindrical2dVertexMesh(Cylindrical2dMesh& rMesh, bool
+                    double ratio = 0.5;
+                    c_vector<double,2> new_node_location = normal_vector + p_node_a->rGetLocation() + ratio*edge;
+                    
+                    // Check if near other nodes (TODO look at this as could be ineficient)
```
Is this todo still relevant?

jmosborne replied:
Removed

fcooper8472 commented on mesh/src/vertex/Cylindrical2dVertexMesh.cpp:
```diff
@@ -150,40 +152,100 @@ Cylindrical2dVertexMesh::Cylindrical2dVertexMesh(Cylindrical2dMesh& rMesh, bool
+                        if (distance < node_clearance)
+                        {   
+                            node_clear = false;
+                            //break;
```
If this break is not required it should be removed.

jmosborne replied:
uncommented it as should be uncommented

fcooper8472 commented on mesh/src/vertex/Cylindrical2dVertexMesh.cpp:
```diff
@@ -150,40 +152,100 @@ Cylindrical2dVertexMesh::Cylindrical2dVertexMesh(Cylindrical2dMesh& rMesh, bool
+            ++elem_iter)
+        {   
+            bool bad_element = false;
+            double edge_threshold = 1.5; //TODO Make setable variable!
```
Agreed, this looks like it should be settable!

jmosborne replied:
I have reworded this and another todo and left as a future change as these chosen varies are sufficient for the current use cases.

fcooper8472 commented on mesh/src/vertex/Cylindrical2dVertexMesh.cpp:
```diff
@@ -364,7 +425,7 @@ void Cylindrical2dVertexMesh::CheckNodeLocation(Node<2>* pNode)
         pNode->rGetModifiableLocation()[0] = x_location + mWidth;
     }
-    else if (x_location > mWidth)
+    else if (x_location >= mWidth)
```
Why has this changed?

jmosborne replied:
The code is 

if (x_location < 0)
    {
        pNode->rGetModifiableLocation()[0] = x_location + mWidth;
    }
    else if (x_location >= mWidth)
    {
        pNode->rGetModifiableLocation()[0] = x_location - mWidth;
    }

So if the second one is < then modes  at mWidth aren't moved to 0 which isn't correct as as here 0=mWidth

Ive updated the hpp to have the correct closed/open interval [0,mWidth)

fcooper8472 commented on mesh/test/mutable/TestCylindrical2dMesh.hpp:
```diff
@@ -325,6 +325,61 @@ class TestCylindrical2dMesh : public CxxTest::TestSuite
         TS_ASSERT_EQUALS(p_mesh->GetNumElements(), 2*cells_across*(cells_up-1));
         TS_ASSERT_EQUALS(p_mesh->GetNumBoundaryElements(), 1u); // boundary elements removed now halo nodes are used
     }
+    /* 
```
formatting: if you run the formatter over these changes the whitespace will be fixed

jmosborne replied:
Fixed

fcooper8472 reviewed (changes requested):
@jmosborne the code looks fine - there are a few todos that you might want to address, and some odd things like commented lines that perhaps should be removed.

Various parts of the code don't conform to our style guide. This is really simple to fix now, because we use clang format (see the .clang-format file in the repository). All modern IDEs will happily auto-format code blocks (or entire files) for you. Should be simple to set up: which IDE are you currently using?

Finally, this was opened in ticket 3089 on trac (in the archive here: https://github.com/Chaste/trac_archive/blob/master/issues/3089.md). Could you open a new issue please, so that it's possible to follow the PR through to a GitHub issue explaining the purpose of the change?

jmosborne committed:
Issue 142 Fixing PR Comments

jmosborne commented:
Issue for this PR is https://github.com/Chaste/Chaste/issues/142.

I fixed the comments. I fixed most formatiing errors manually as when i did it automatically 50% of lines in the files had changes. all formatting is correct in new code (and some old code) so its a net positive change!

jmosborne committed:
Issue 142 Tidying TODOs

jmosborne committed:
Issue 142 Tidying more formatting

jmosborne committed:
Formatting edits to restart builds

jmosborne committed:
Merge branch 'develop' into ticket_3089
